src/main/java/org/broadinstitute/hellbender/utils/read/MutableGATKRead.java

```diff
@@ -0,0 +1,76 @@
+package org.broadinstitute.hellbender.utils.read;
+
+import htsjdk.samtools.Cigar;
+import htsjdk.samtools.util.Locatable;
+
+/**
+ * Unified read interface for use throughout the GATK.
+ *
+ * Includes operations that allow reads to be modified in-place. Use {@link @GATKRead} if you want immutability.
+ */
+public interface MutableGATKRead extends GATKRead {
```

Comment:
I think you should add a comment stating that this interface is not intended to be implemented outside of the GATK core (ie clients are not supposed to implement it)